### PR TITLE
update ed25519 address scheme

### PR DESCRIPTION
**Pull request by ebuchman**
Ref #103

### CODE REVIEW

File: pub_key.go [hunk 1]
```diff
@@ -38,11 +43,9 @@ var _ PubKey = PubKeyEd25519{}
 // Implements PubKeyInner
 type PubKeyEd25519 [32]byte
 
+// Address is the SHA256-20 of the raw pubkey bytes.
 func (pubKey PubKeyEd25519) Address() Address {
-	// append type byte
-	hasher := ripemd160.New()
-	hasher.Write(pubKey.Bytes()) // does not error
-	return Address(hasher.Sum(nil))
+	return Address(tmhash.Sum(pubKey[:]))
```
Comment (liamsi) [inline pub_key.go:48]: Just to make sure, as we dropped ed25519 support for wallets, we still need the `Address()` of a `PubKeyEd25519` for other purposes, right?

**Review by liamsi — APPROVED** [review]
LGTM. Left one question in the comments.

File: pub_key.go (hunk 1 above)
Comment (ethanfrey) [reply pub_key.go:48]: Huh? Just saw this tidbit here...
Is that true? Using only secp? It would be good to add that to BreakingChanges...

Comment (ebuchman) [reply pub_key.go:48]: will still be supported in go-crypto, we're just not going to have tooling for it for user keys at launch

**Pull request merged**